openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -0,0 +1,646 @@
+        """
+        % (
+            openupgrade.get_legacy_name("journal_id"),
+            openupgrade.get_legacy_name("name"),
+            openupgrade.get_legacy_name("payment_date"),
+        )
+    )
+    for p_id, p_journal_id, p_name, p_payment_date in env.cr.fetchall():
+        p_data[p_id] = {
+            "journal_id": p_journal_id,
+            "name": p_name,
+            "payment_date": p_payment_date,
+        }
+    payments = env["account.payment"].browse(list(p_data.keys()))
+    for payment in payments.with_context(check_move_validity=False):
+        journal = env["account.journal"].browse(p_data[payment.id]["journal_id"])
+        move = env["account.move"].create(
+            {
+                "name": "/",
+                "date": p_data[payment.id]["payment_date"],
+                "payment_id": payment.id,
+                "move_type": "entry",
+                "journal_id": journal.id,
+                "company_id": journal.company_id.id,
+                "currency_id": journal.currency_id.id
+                or journal.company_id.currency_id.id,
+            }
+        )
+        payment.move_id = move
+        deprecated_accounts = env["account.account"].search(
+            [("deprecated", "=", True), ("company_id", "=", payment.company_id.id)]
+        )
+        deprecated_accounts.deprecated = False
+        try:
+            payment._synchronize_to_moves(
+                [
+                    "date",
+                    "amount",
+                    "payment_type",
+                    "partner_type",
+                    "payment_reference",
+                    "is_internal_transfer",
+                    "currency_id",
+                    "partner_id",
+                    "destination_account_id",
+                    "partner_bank_id",
+                    "journal_id",
+                ]
+            )
+        except Exception as e:
+            _logger.error("Failed for payment with id %s: %s", payment.id, e)
+            raise
+        deprecated_accounts.deprecated = True
+        to_write = {
+            "line_ids": [
+                (0, 0, line_vals)
+                for line_vals in payment._prepare_move_line_default_vals(
+                    write_off_line_vals=False
+                )
+            ]
+        }
+        payment.move_id.with_context(skip_account_move_synchronization=True).write(
+            to_write
+        )
+
+
+def try_delete_noupdate_records(env):
+    openupgrade.delete_records_safely_by_xml_id(
+        env,
+        [
+            "account.sequence_payment_customer_invoice",
+            "account.sequence_payment_customer_refund",
+            "account.sequence_payment_supplier_invoice",
+            "account.sequence_payment_supplier_refund",
+            "account.sequence_payment_transfer",
+        ],
+    )
+
+
+def fill_account_move_amounts(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line
```

Comment:
It seems that this is not always a good solution.

Entries related to difference of exchange rates seems to have debit+credit > 0, but amount_currency should stay at 0.

Maybe this should be applicable only for move_type != entry.

I will investigate more.

Reply:
I just added the `move_type != entry`.